src/test/java/DecodeTest.kt

```diff
@@ -0,0 +1,52 @@
+import Decode
+import org.junit.jupiter.api.Test
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+
```

Comment:
There is no unit test which tests the correctness of decoding, as in the positive outcome (without exceptions)